src/librustc_target/abi/mod.rs

```diff
@@ -608,6 +623,15 @@ pub struct Scalar {
     pub valid_range: RangeInclusive<u128>,
 }
 
+impl<CTX> HashStable<CTX> for Scalar {
```

Comment:
You could use a derive macro for `Scalar` here too with a `RangeInclusive` impl.